src/components-styled/title-with-icon.tsx

```diff
@@ -0,0 +1,66 @@
+import css from '@styled-system/css';
+import React, { useMemo } from 'react';
+import { Box } from './base';
+import { Heading } from './typography';
+
+interface IProps {
+  title: string;
+  Icon: React.ReactNode;
```

Comment:
```suggestion
  icon: React.ReactNode;
```
I think what you're passing in is not a component constructor but an instance because <Icon/> in the calling context will already do the instantiation.

Reply:
Also make it React.FunctionComponent instead of ReactNode, because node can be anything including a plain string.

Reply:
Fixed, it's come to `JSX.Element`, only because the way icons are imported (`import GetestIcon from '~/assets/test.svg';`), the compiler only allows me to use this one.